<!DOCTYPE html>
<html>

  <head>
    <meta charset='utf-8'>
    <meta http-equiv="X-UA-Compatible" content="chrome=1">
    <meta name="description" content="A NGS data analysis course : An introductory course about NGS data analysis and visualization">

    <link rel="stylesheet" type="text/css" media="screen" href="stylesheets/stylesheet.css">

    <title>A NGS data analysis course</title>
  </head>

  <body>

    <!-- HEADER -->
    <div id="header_wrap" class="outer">
        <header class="inner">
          <a id="forkme_banner" href="https://github.com/ngs-course">View on GitHub</a>

          <h1 id="project_title">A NGS data analysis course</h1>
          <h2 id="project_tagline">An introductory course about NGS data analysis and visualization</h2>

        </header>
    </div>

    <!-- MAIN CONTENT -->
    <div id="main_content_wrap" class="outer">
      <section id="main_content" class="inner">
        <h1>
<a id="overview" class="anchor" href="#overview" aria-hidden="true"><span class="octicon octicon-link"></span></a>Overview</h1>

<p>Welcome to this introductory course about <em>NGS data analysis and visualization</em>. During this course you will learn the basics of NGS data analysis and visualization in a Linux environment, current used software and best practices will be explained. This course is focused in NGS data alignment for both DNA and RNA (RNA-seq), variant calling, differential expression analysis and data visualization.</p>

<p>This is course is scheduled for a 3 days and assumes a very basic knowledge of NGS data analysis and Linux. All materials in this is course are free and open, feel free to reuse them as you want. All the data for the tutorials are available in our <a href="https://www.dropbox.com/sh/4qkqch7gyt888h7/AABD_i9ShwryfAqGeJ0yqqF3a">DropBox folder</a>.</p>

<h1>
<a id="schedule" class="anchor" href="#schedule" aria-hidden="true"><span class="octicon octicon-link"></span></a>Schedule</h1>

<h3>
<a id="day-1" class="anchor" href="#day-1" aria-hidden="true"><span class="octicon octicon-link"></span></a>Day 1</h3>

<p>During this first day we will start with an <em>introduction to NGS technology</em> and in the data preparation for their analysis. You will also learn the basics of the <em>GNU/Linux shell introduction</em>. At this point we are ready to start preparing and working with the data: <em>Quality Control</em> (QC) and <em>NGS data alignment</em>.</p>

<h5>
<a id="course-presentation" class="anchor" href="#course-presentation" aria-hidden="true"><span class="octicon octicon-link"></span></a>Course Presentation</h5>

<ul>
<li>Presentation [<a href="Course_Materials/presentation/presentation_mda15.pdf">PDF</a> | <a href="Course_Materials/presentation/presentation_mda15.odp">OpenOffice</a>]</li>
</ul>

<h5>
<a id="genomic-data-analysis-overview" class="anchor" href="#genomic-data-analysis-overview" aria-hidden="true"><span class="octicon octicon-link"></span></a>Genomic Data Analysis Overview</h5>

<ul>
<li>Presentation <a href="Course_Materials/course_overview/Dopazo_Intro_Cambridge.pdf">PDF</a>
</li>
</ul>

<h5>
<a id="introduction-to-ngs-technologies" class="anchor" href="#introduction-to-ngs-technologies" aria-hidden="true"><span class="octicon octicon-link"></span></a>Introduction to NGS technologies</h5>

<p>Very brief overview about NGS technologies and some concepts.</p>

<ul>
<li><a href="Course_Materials/intro-ngs/ngs_introduction_mda15.pdf">Presentation</a></li>
</ul>

<h5>
<a id="introduction-gnulinux-shell" class="anchor" href="#introduction-gnulinux-shell" aria-hidden="true"><span class="octicon octicon-link"></span></a>Introduction GNU/Linux shell</h5>

<ul>
<li>Presentation [<a href="Course_Materials/intro-linux/intro_Linux_mda15.pdf">PDF</a> | <a href="Course_Materials/intro-linux/intro_Linux_mda15.odp">OpenOffice</a>]</li>
</ul>

<h5>
<a id="quality-control-for-ngs-raw-data-fastq" class="anchor" href="#quality-control-for-ngs-raw-data-fastq" aria-hidden="true"><span class="octicon octicon-link"></span></a>Quality control for NGS raw data (FASTQ)</h5>

<ul>
<li>
<a href="Course_Materials/quality_control/presentation/quality_control_presentation.pdf">Presentation</a> - <a href="Course_Materials/quality_control/tutorial/quality_control.html">Tutorial</a> - <a href="https://www.dropbox.com/sh/4qkqch7gyt888h7/AAAqebBSC6JgDGq4emwNORCaa/quality_control">Data</a>
</li>
</ul>

<h5>
<a id="dna-and-rna-seq-ngs-read-alignment" class="anchor" href="#dna-and-rna-seq-ngs-read-alignment" aria-hidden="true"><span class="octicon octicon-link"></span></a>DNA and RNA-seq NGS read alignment</h5>

<ul>
<li>Presentation [<a href="Course_Materials/alignment/presentation/ngs-read-mapping-imedina-mda15.pdf">PDF</a> | <a href="Course_Materials/alignment/presentation/ngs-read-mapping-imedina-mda15.odp">OpenOffice</a>] - <a href="Course_Materials/alignment/tutorial/example.html">Tutorial</a>
</li>
</ul>

<h3>
<a id="day-2" class="anchor" href="#day-2" aria-hidden="true"><span class="octicon octicon-link"></span></a>Day 2</h3>

<p>During the second day we will focus on <em>Data Visualization</em> of DNA and RNA-seq alignments and <em>Variant Calling and prioritization</em> together with the visualization of VCF files. </p>

<h5>
<a id="ngs-data-visualization" class="anchor" href="#ngs-data-visualization" aria-hidden="true"><span class="octicon octicon-link"></span></a>NGS data Visualization</h5>

<ul>
<li>
<a href="Course_Materials/visualization/presentation/2015-Cambridge_visualisation.pdf">Presentation</a> - <a href="Course_Materials/visualization/tutorial/000_example.html">Guided example</a> - <a href="Course_Materials/visualization/tutorial/010_example.html">Tutorial 1</a> - <a href="Course_Materials/visualization/tutorial/020_example.html">Tutorial 2</a>
</li>
</ul>

<h5>
<a id="variant-calling-analysis" class="anchor" href="#variant-calling-analysis" aria-hidden="true"><span class="octicon octicon-link"></span></a>Variant calling analysis</h5>

<ul>
<li>
<a href="Course_Materials/variant_calling/presentation/2015-Cambridge_variant_calling.pdf">Presentation</a> - <a href="Course_Materials/variant_calling/tutorial/010_example.html">Tutorial 1</a> - <a href="Course_Materials/variant_calling/tutorial/020_example.html">Tutorial 2</a> - <a href="Course_Materials/variant_calling/tutorial/030_example.html">Tutorial 3</a>
</li>
</ul>

<h5>
<a id="variant-annotation" class="anchor" href="#variant-annotation" aria-hidden="true"><span class="octicon octicon-link"></span></a>Variant annotation</h5>

<ul>
<li>
<a href="Course_Materials/variant_annotation/presentation/2015-Cambridge_variant_annotation.pdf">Presentation</a> - <a href="Course_Materials/variant_annotation/tutorial/annovar.html">Annovar</a> - <a href="Course_Materials/variant_annotation/tutorial/cellbase.html">CellBase</a>
</li>
</ul>

<h5>
<a id="some-considerations-on-candidate-variant-prioritization" class="anchor" href="#some-considerations-on-candidate-variant-prioritization" aria-hidden="true"><span class="octicon octicon-link"></span></a>Some considerations on candidate variant prioritization</h5>

<ul>
<li><a href="Course_Materials/variant_prioritization/presentation/Dopazo-Prioritization_Cambridge.pdf">Presentation</a></li>
</ul>

<h5>
<a id="variant-prioritization" class="anchor" href="#variant-prioritization" aria-hidden="true"><span class="octicon octicon-link"></span></a>Variant prioritization</h5>

<ul>
<li>
<a href="Course_Materials/variant_prioritization/presentation/2015-Cambridge_variant_prioritization.pdf">Presentation</a> - <a href="https://www.dropbox.com/sh/4qkqch7gyt888h7/AADPzrs9NGg0PjVqnwQocUJUa/annotation/hpg-variant/examples">Data</a>
</li>
</ul>

<h3>
<a id="day-3" class="anchor" href="#day-3" aria-hidden="true"><span class="octicon octicon-link"></span></a>Day 3</h3>

<p>In the last day we will focus on RNA-seq data analysis and functional interpretation.</p>

<h5>
<a id="association-studies" class="anchor" href="#association-studies" aria-hidden="true"><span class="octicon octicon-link"></span></a>Association studies</h5>

<ul>
<li>
<a href="Course_Materials/association_studies/presentation/association_studies_presentation.pdf">Presentation</a> - <a href="Course_Materials/association_studies/tutorial/association_studies.html">Tutorial</a>
</li>
</ul>

<h5>
<a id="rna-seq-data-analysis" class="anchor" href="#rna-seq-data-analysis" aria-hidden="true"><span class="octicon octicon-link"></span></a>RNA-Seq data analysis</h5>

<ul>
<li>
<a href="Course_Materials/rna_seq/presentation/rna_seq_presentation.pdf">Presentation</a> - <a href="Course_Materials/rna_seq/tutorial/rna_seq.html" title="Cufflinks">Tutorial 1</a> - <a href="Course_Materials/rna_seq/tutorial_htseq/rna_seq_htseq.html" title="HTSeq">Tutorial 2</a> - <a href="https://www.dropbox.com/sh/4qkqch7gyt888h7/AAB77HfUPkiXBj1MmshjyKtYa/rna_seq">Data</a><br>
You will also need the <a href="https://www.dropbox.com/s/3ci9yuh2hh9ck8g/f000_chr21_ref_genome_sequence.fa?dl=0">reference genome</a> and the <a href="https://www.dropbox.com/s/4r3tjwu840i1iio/f005_chr21_genome_annotation.gtf?dl=0">GTF file with the annotation</a>
</li>
</ul>

<h5>
<a id="functional-analysis" class="anchor" href="#functional-analysis" aria-hidden="true"><span class="octicon octicon-link"></span></a>Functional analysis</h5>

<ul>
<li>
<a href="Course_Materials/functional/presentation/babelomics_data_analysis.pdf">Presentation</a> - <a href="http://bioinfo.cipf.es/babelomicstutorial/">Tutorial</a>
</li>
</ul>

<h5>
<a id="big-data-analysis-and-visualization" class="anchor" href="#big-data-analysis-and-visualization" aria-hidden="true"><span class="octicon octicon-link"></span></a>Big Data analysis and visualization</h5>

<ul>
<li><a href="Course_Materials/big_data/big_data_presentation.pdf">Presentation</a></li>
</ul>

<hr>

<h1>
<a id="about" class="anchor" href="#about" aria-hidden="true"><span class="octicon octicon-link"></span></a>About</h1>

<p>This course is usually carried out by experienced researchers from CIPF, EMBL-EBI and University of Cambridge. You can ask any question to David Montaner (<a href="mailto:dmontaner@cipf.es">dmontaner@cipf.es</a>), Marta Bleda (<a href="mailto:mb2033@cam.ac.uk">mb2033@cam.ac.uk</a>), Ignacio Medina (<a href="mailto:im411@cam.ac.uk">im411@cam.ac.uk</a>, <a href="mailto:imedina@ebi.ac.uk">imedina@ebi.ac.uk</a>), Javier Lopez (<a href="mailto:fjlopez@ebi.ac.uk">fjlopez@ebi.ac.uk</a>) and Joaquin Dopazo (<a href="mailto:jdopazo@cip.es">jdopazo@cip.es</a>).</p>

<hr>

<p></p>
<a href="http://www.ngscourse.org/">www.ngscourse.com</a>

      </section>
    </div>

    <!-- FOOTER  -->
    <div id="footer_wrap" class="outer">
      <footer class="inner">
        <p>Published with <a href="https://pages.github.com">GitHub Pages</a></p>
      </footer>
    </div>

    

  </body>
</html>
